content/post/24-Flux-iOS.md

```diff
@@ -0,0 +1,515 @@
+2. Implement a handler that will be called for each of the individual actions.
+3. Within the handler, perform the necessary business logic and update the state upon completion.
+
+As a concrete example we can take a look at how the `AnnotationFilterStore` handles the `toggleFilterAction`:
+
+{{< highlight swift >}}
+func handleToggleFilterAction(toggleFilterAction: AnnotationFilteringActions.ToggleFilterAction) {
+    var filter = toggleFilterAction.filter
+    filter.enabled = !filter.enabled
+
+    // Check for issue specific filters
+    if filter is IssueAssignedToFilter ||
+        filter is IssueStatusAnnotationFilter ||
+        filter is IssueAssignedToFilter ||
+        filter is IssueUnassignedFilter {
+            // if no annotation types are filtered, activate the issue/punchItem type
+            var issueTypeFilter = self._annotationFilterService.annotationTypeFilterGroup.issueTypeFilter
+            if self._annotationFilterService.annotationTypeFilterGroup.activeFilterCount() == 0 ||
+                issueTypeFilter?.enabled == false {
+                    issueTypeFilter?.enabled = true
+            }
+    }
+
+    self._applyFilter()
+}
+{{< /highlight >}}
+
+This example is purposefully not simplified. So let's break it down a litte. The `handleToggleFilterAction` is invoked whenever a `ToggleFilterAction` is dispatched.
+The `ToggleFilterAction` carries information about which specific filter should be toggled.
+
+As a very first step of implementing this business logic, the method simply toggles the filter by toggling the value of the `filter.enabled`.
+
+Then we implement some custom business logic for this feature. When working with filters that are intended to filter issue annotations we have some cases in which we need to activate the `issueTypeFilter`. It wouldn't make sense to dive into the specifics of this PlanGrid feature, but the idea is that this method encapsulates any business logic related to toggling filters.
+
+At the end of the method we're calling the `_applyFilter()` method. This is a shared method that is used by multiple action handlers:
+
+{{< highlight swift >}}
+func _applyFilter() {
+    self._annotationFilterService.applyFilter()
+
+    self._state.value?.isFiltering = self._annotationFilterService.allFilterGroups.reduce(false) { isFiltering, filterGroup in
+        isFiltering || (filterGroup.activeFilterCount() > 0)
+    }
+
+    // Phantom state update to refresh the cell state, technically not needed since filters are reference types
+    // and previous statement already triggers a state update.
+    self._state.value = self._state.value
+}
+{{< /highlight >}}
+
+The call to `self._annotationFilterService.applyFilter()` is the one that actually triggers the filtering of annotations that are displayed on a sheet. The filtering logic itself is somewhat complex, so it makes sense to move this into a separate, dedicated type.
```

Comment:
I guess my question wasn't clear enough, I thought puristic implementation would make `AnnotationFilterService` to observe some sort of `shouldApplyFilter` boolean property on a global state object, and would trigger `applyFilter()` function in response to that boolean property flipped to `true` - however that seems like a lot of overhead to trigger filtering. 
Thanks for the explanation though - this approach does make things simpler.